Source/KSCrash/Source/KSCrash/Recording/BSG_KSCrash.m

```diff
@@ -253,6 +258,20 @@ - (BOOL)install {
     return true;
 }
 
+/**
+ * Set up listeners for un/loaded frameworks.  Maintaining our own list of framework Mach
+ * headers means that we avoid potential deadlock situations where we try and suspend
+ * lock-holding threads prior to loading mach headers as part of our normal event handling
+ * behaviour.
+ */
+- (void)listenForLoadedBinaries {
+    bsg_initialise_mach_binary_headers();
+    _dyld_register_func_for_add_image(&bsg_mach_binary_image_added);
+    _dyld_register_func_for_remove_image(&bsg_mach_binary_image_removed);
```

Comment:
Potentially worth registering `bsg_mach_binary_image_removed` first. This avoids the case where some binary image is removed on another thread _while_ the `bsg_mach_binary_image_added` callback is executed for the first time.